_posts/2025-03-12-superfast-debezium.adoc

```diff
@@ -0,0 +1,719 @@
+
+[id="why"]
+== Introduction
+One way to use Debezium is by embedding it in a web application along with a connector to listen for changes in a data source. What's particularly interesting is that, with some tweaks, it's possible to build a CDC system with ultra-fast startup times and a minimal memory footprint by leveraging Quarkus and GraalVM. We will ingest data from a postgres database into a native Quarkus application with Debezium that apply an easy https://debezium.io/documentation/reference/stable/transformations/applying-transformations-selectively.html[transformation].
+
+[id="why"]
+=== Why
+I'd say, why not? From another perspective, the CDC pattern operates within the realm of distributed systems, which must adhere to principles like scalability, fault tolerance, and high availability. Optimizing resource usage not only lowers costs but also helps reduce your carbon footprint when you are trying to scale the application.
+
+[id="disclaimer"]
+=== Disclaimer
+The tweaks suggested here serve solely as a **proof of concept** to demonstrate Debezium's capabilities. They are not intended for production use but rather to showcase its potential. Debezium and its surrounding third-party libraries are not designed to be fully native for all use cases. Instead, the tweaks presented here focus on a specific **scenario**.
+
+[id="side_effect"]
+=== The side effect
+This guide can also help you understand how to make a Java application native. However, the best and most reliable approach is to use only dependencies provided by Quarkus.
+
+[id="considerations"]
+=== Some Considerations about GraalVM
+
+GraalVM compiles Java applications into native binaries using **Ahead-of-Time** (AOT) compilation. This process involves analyzing the entire application to identify all reachable code, including classes, methods, and dependencies. However, one widely used Java feature *—reflection—* introduces dynamic behavior that cannot always be resolved at build time, often leading to **runtime failures** such as `ClassNotFoundException`. Additionally, many libraries rely on **dynamic proxy generation**, which is not supported in AOT compilation. This is where Quarkus comes to the rescue, providing a set of native-ready extensions and libraries to overcome these limitations.
+
+[id="build"]
+== Build the Basic Application
+
+Alright, let’s start building the application! All the proposed code is available on https://github.com/debezium/debezium-examples/quarkus-native[GitHub].
```

Comment:
Before publish it's necessary to merge the example

related to: https://github.com/debezium/debezium-examples/pull/374

Reply:
It really won't matter which you do first, it takes about 5-10 minutes before the blog shows up on the website, so in that time you can merge the example or do the example first.  Your call @kmos

Reply:
ok, if @jpechane is [ok](https://github.com/debezium/debezium-examples/pull/374#pullrequestreview-2685146836), I can merge both